TUnit.Assertions/AssertConditions/AssertionResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace TUnit.Assertions.AssertConditions;
+
+public class AssertionResult
+{
+    public bool IsPassed { get; }
+    public string? Message { get; }
+
+    private AssertionResult(bool isPassed, string? message)
+    {
+        IsPassed = isPassed;
+        Message = message;
+	}
+
+	public static AssertionResult FailIf(Func<bool> isFailed, string message)
```

Comment:
Does this need to take a func? Looks like it invokes it immediately. Couldn't you just pass in the boolean result directly?

Reply:
I could probably get rid of it here, but not in the `OrFailIf` method, as it would then throw, e.g. in 
```csharp
		=> AssertionResult
			.FailIf(
				() => actualValue is null,
				$"{ActualExpression ?? typeof(TActual).Name} is null")
			.OrFailIf(
				() => actualValue.Contains(inner, equalityComparer),
				"it was found in the collection");
```
because both conditions are always evaluated and not just when all previous conditions didn't fail.
However having different method signatures for `FailIf` and `OrFailIf` looks strange to me...